posts/MarcoB/2024-07-12-embeddable-inheritance.adoc

```diff
@@ -0,0 +1,237 @@
+= Hibernate ORM 6.6 - Embeddable Inheritance
+Marco Belladelli
+:awestruct-tags: [ "Hibernate ORM", "Discussions" ]
+:awestruct-layout: blog-post
+---
+
+The first release candidate for Hibernate ORM `6.6` has already been released, and a final version will be available very soon. In today's blog post, I'm going to delve into a feature that has been requested for a long time but has never made it into our framework until now: Embeddable Inheritance.
+
+[[embeddable-values]]
+== What are Embeddable values
+
+Embeddable properties have been a part of Hibernate for a long time, but let me give you a quick rundown of their basic functionalities. If you already know all about embeddables and just want to hear the news, feel free to skip to the <<embeddable-inheritance,next section>>.
+
+Embeddable types, identified by the `@Embeddable` annotation, are a *composition of values* that, unlike entities, do not directly correspond to database tables. Their state is tied to the entities they're used in as `@Embedded` properties, and they are *reusable* across different mappings. Embeddables can also be used as _composite primary keys_ through the `@EmbeddedId` or `@IdClass` annotations, and as `@ElementCollection` values.
+
+Here's an example of a very simple embeddable type consisting of two properties:
+
+====
+[source, java, indent=0]
+----
+@Embeddable
+class Animal {
+	private int age;
+
+	private String name;
+}
+----
+====
+
+And here's an entity mapping that uses it:
+
+====
+[source, java, indent=0]
+----
+@Entity
+class Owner {
+	@Id
+	private Long id;
+
+	@Embedded
+	private Animal pet;
+}
+----
+====
+
+The database table corresponding to the `Owner` entity will include both the properties for the `Animal` embeddable:
+
+====
+[source, sql, indent=0]
+----
+    create table Owner (
+        id bigint not null,
+        age integer,
+        name varchar(255),
+        primary key (id)
+    )
+----
+====
+
+Note that embeddables can be made up of basic values as well as associations. For more in-depth information about all the possibilities embeddables offer, you can refer to link:https://docs.jboss.org/hibernate/orm/current/userguide/html_single/Hibernate_User_Guide.html#embeddables[this chapter] of the Hibernate user guide.
```

Comment:
I usually like to use `current` so that if we add / change the contents of that chapter in the future the blog post still points to correct information, but I guess it'll be long forgotten by that :grin: